specification/synapse/resource-manager/Microsoft.Synapse/stable/2020-12-01/sqlServer.json

```diff
@@ -1331,6 +1569,91 @@
         }
       }
     },
+    "EncryptionProtectorListResult": {
+      "description": "A list of server encryption protectors.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/EncryptionProtector"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "EncryptionProtectorProperties": {
+      "description": "Properties for an encryption protector execution.",
+      "required": [
+        "serverKeyType"
+      ],
+      "type": "object",
+      "properties": {
+        "subregion": {
+          "description": "Subregion of the encryption protector.",
+          "type": "string",
+          "readOnly": true
+        },
+        "serverKeyName": {
+          "description": "The name of the server key.",
+          "type": "string"
+        },
+        "serverKeyType": {
+          "description": "The encryption protector type like 'ServiceManaged', 'AzureKeyVault'.",
+          "enum": [
+            "ServiceManaged",
+            "AzureKeyVault"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ServerKeyType",
+            "modelAsString": true
+          }
+        },
+        "uri": {
+          "description": "The URI of the server key.",
+          "type": "string",
+          "readOnly": true
+        },
+        "thumbprint": {
+          "description": "Thumbprint of the server key.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "EncryptionProtector": {
+      "description": "The server encryption protector.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v1/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "description": "Kind of encryption protector. This is metadata used for the Azure portal experience.",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Resource location.",
```

Comment:
why is location a readonly property?

Reply:
This is a partner (SQL) API that we're proxying the request to. Hence the swagger is actually dependant on that and we've ported the same swagger posted on the SQL side to Synapse. 

Having said this, I will take this up with the SQL folks, see if there is a code dependency on this property and update the Swagger accordingly. However, considering this is a readonly response property, I would like to take this up in the next Swagger update iteration if that's ok.